lib/membrane_timescaledb_reporter/models/model.ex

```diff
@@ -2,58 +2,80 @@ defmodule Membrane.Telemetry.TimescaleDB.Model do
   require Logger
   import Ecto.Query
   alias Membrane.Telemetry.TimescaleDB.Repo
-  alias Membrane.Telemetry.TimescaleDB.Model.{Metric, ElementPath}
+  alias Membrane.Telemetry.TimescaleDB.Model.{Measurement, ElementPath}
 
-  def create_all_metrics(metrics) do
+  # inserts given element paths if they don't exist and returns them, otherwise does nothing
+  defp insert_all_element_paths(multi, element_paths) do
+    Ecto.Multi.insert_all(multi, :insert_all_element_paths, ElementPath, element_paths,
+      on_conflict: :nothing,
+      returning: true
+    )
+  end
+
+  # fetches remaining paths that have not been returned by insert_all_element_paths/2 and bundles them all together
+  defp fetch_remaining_paths(multi, element_paths) do
+    Ecto.Multi.run(multi, :fetch_remaining_paths, fn repo, changes ->
+      %{
+        insert_all_element_paths: {_n, inserted_element_paths}
+      } = changes
+
+      fetched_paths =
+        inserted_element_paths
+        |> Enum.map(&%{path: &1.path})
+        |> MapSet.new()
```

Comment:
There's `Map.new/2` that you can use here:
```suggestion
        |> MapSet.new(&Map.take(&1, [:path]))
```